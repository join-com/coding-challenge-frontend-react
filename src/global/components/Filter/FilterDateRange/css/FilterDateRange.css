.FilterDateRange__label {
  display: block;
  font-size: 1.5rem;
  font-weight: bold;
}

.FilterDateRange__input {
  margin-top: calc(var(--space) / 3);
  display: block;
  font-family: inherit;
  font-weight: normal;
  padding: calc(var(--element-padding) / 1.4);
}

.FilterDateRange__input[placeholder] {
  color: var(--color-dark-grey);
}
