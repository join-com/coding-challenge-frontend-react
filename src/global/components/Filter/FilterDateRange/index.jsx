import React from 'react';

import PropTypes from 'prop-types';
import filterType from '../types/filter';

import { bemCls } from '../../../utils/ClassNameHelpers';

import InputDate from '../../Inputs/InputDate';

import './css/FilterDateRange.css';

const FilterDateRange = ({
  filter, value, className, handleFiltersChange,
}) => {
  if (filter) {
    const {
      name, type, label, placeholder,
    } = filter;

    return (
      <div className={`FilterDateRange ${className}`}>
        <InputDate
          className={`FilterDateRange ${className} ${
            bemCls(`FilterDateRange ${className}`, `__${name}`)}`}
          name={name}
          type={type}
          label={label}
          placeholder={placeholder}
          value={value}
          onChange={handleFiltersChange}
        />
      </div>
    );
  }
  return null;
};

export default FilterDateRange;

FilterDateRange.propTypes = {
  filter: filterType,
  value: PropTypes.string,
  className: PropTypes.string,
  handleFiltersChange: PropTypes.func,
};

FilterDateRange.defaultProps = {
  filter: null,
  value: '',
  className: '',
  handleFiltersChange: () => {},
};
