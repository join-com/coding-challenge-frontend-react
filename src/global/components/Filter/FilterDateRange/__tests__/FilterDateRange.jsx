import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import FilterDateRange from '../index';

const props = {
  filter: {
    name: 'myPropertyName',
    type: 'date',
    from: 'myParamNameFrom',
    to: 'myParamNameTo',
    format: 'YYYY-MM-DD',
    labelFrom: 'MyFilterInputLabel From',
    labelTo: 'MyFilterInputLabel To',
    placeholderFrom: 'My Filter Input Placeholder From',
    placeholderTo: 'My Filter Input Placeholder To',
  },
  value: {
    from: 1543309200,
    to: 1553309200,
  },
  className: 'MyFilterClassname',
  handleFiltersChange: () => {},
};

describe('FilterDateRange', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<FilterDateRange {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
