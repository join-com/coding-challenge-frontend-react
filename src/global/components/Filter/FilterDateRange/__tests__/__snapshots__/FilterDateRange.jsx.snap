// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDateRange should render the component correctly 1`] = `
<div
  className="FilterDateRange MyFilterClassname"
>
  <InputDateWrapper
    className="FilterDateRange MyFilterClassname FilterDateRange__myPropertyNameFrom MyFilterClassname__myPropertyNameFrom"
    label="MyFilterInputLabel From"
    name="myPropertyNameFrom"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="My Filter Input Placeholder From"
    type="date"
    value={1543309200}
  />
  <InputDateWrapper
    className="FilterDateRange MyFilterClassname FilterDateRange__myPropertyNameTo MyFilterClassname__myPropertyNameTo"
    label="MyFilterInputLabel To"
    name="myPropertyNameTo"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="My Filter Input Placeholder To"
    type="date"
    value={1553309200}
  />
</div>
`;
